packages/runtime/runtime-utils/src/summaryUtils.ts

```diff
@@ -203,3 +206,80 @@ export function convertToSummaryTree(
         return builder.getSummaryTree();
     }
 }
+
+/**
+ * Converts ISnapshotTree to ISummaryTree format and tracks stats.
+ * @param snapshot - snapshot in ISnapshotTree format
+ */
+export function convertSnapshotTreeToSummaryTree(
+    snapshot: ISnapshotTree,
+): ISummaryTreeWithStats {
+    const blobMapInitial = new Map(Object.entries(snapshot.blobs));
+    const blobMapFinal = new Map<string, string>();
+    for (const [key, value] of blobMapInitial.entries()) {
+        if (blobMapInitial.has(value)) {
+            blobMapFinal[key] = blobMapInitial.get(value);
+        }
+    }
+
+    const builder = new SummaryTreeBuilder();
+    for (const [key, value] of Object.entries(blobMapFinal)) {
+        const decoded = fromBase64ToUtf8(value);
+        builder.addBlob(key, decoded);
+    }
+    for (const [key, tree] of Object.entries(snapshot.trees)) {
+        const subtree = convertSnapshotTreeToSummaryTree(tree);
+        builder.addWithStats(key, subtree);
+    }
```

Comment:
Maybe add an `assert` that there are no commit tree entries here?

Reply:
Can the snapshot tree not have commit trees ever?

Reply:
They shouldn't anymore, but the main reason for the assert is that even if they _did_ have a commit, we aren't handling it in this convert function, so our generated summary tree will be missing that.